helix-core/src/main/java/org/apache/helix/controller/rebalancer/util/DelayedRebalanceUtil.java

```diff
@@ -279,4 +287,157 @@ public static void setRebalanceScheduler(String resourceName, boolean isDelayedR
       }
     }
   }
+
+  /**
+   * Computes the partition replicas that needs to be brought up to satisfy minActiveReplicas while downed instances
+   * are within the delayed window.
+   * Keep all current assignment with their current allocation.
+   * @param clusterData Cluster data cache.
+   * @param replicaMap A set of assignable replicas by resource name.
+   * @param liveEnabledInstances The set of live and enabled instances.
+   * @param currentAssignment Current assignment by resource name.
+   * @param allocatedReplicas The map from instance name to assigned replicas, the map is populated in this method.
+   * @return The replicas that need to be assigned.
+   */
+  public static Set<AssignableReplica> findToBeAssignedReplicasForMinActiveReplica(
+      ResourceControllerDataProvider clusterData,
+      Map<String, Set<AssignableReplica>> replicaMap,
+      Set<String> liveEnabledInstances,
+      Map<String, ResourceAssignment> currentAssignment,
+      Map<String, Set<AssignableReplica>> allocatedReplicas) {
+    Map<String, List<String>> partitionsMissingMinActiveReplicas =
+        findPartitionsMissingMinActiveReplica(clusterData, currentAssignment);
+    if (partitionsMissingMinActiveReplicas.isEmpty()) {
+      return Collections.emptySet();
+    }
+    Set<AssignableReplica> toBeAssignedReplicas = new HashSet<>();
+
+    for (String resourceName : replicaMap.keySet()) {
```

Comment:
note: Since we are using partitionsMissingMinActiveReplicas, why are we iterating again?

Reply:
As the comment `keep all current assignment and add to allocated replicas`, we should be populating `allocatedReplicas` to preserve current assignment, and this requires going through all resources -- not just the resource-partition missing minActiveReplica

Let me add more comments to clarify, thanks